packages/app/src/hooks/blockchain/useSapienceWriteContract.ts

```diff
@@ -455,6 +476,125 @@ export function useSapienceWriteContract({
         // Validate and switch chain if needed
         await validateAndSwitchChain(_chainId);
 
+        // If using ZeroDev session key, use smart account for batched transactions
+        // For Arbitrum (42161), use Arbitrum-specific session check
+        const useSessionForThisChain =
+          _chainId === 42161
+            ? shouldUseArbitrumSessionKey
+            : shouldUseSessionKey;
```

Comment:
need a different ternary condition for ethereal - don't use sessions at all for now while we don't have a bundler, otherwise it is Undefined Behaviour when submitting a tx (no guarantee that our codebase will still work as intended)